plugins/repo/dig/plugin.py

```diff
@@ -41,33 +44,128 @@ def parseOutputString(self, output):
+
+
+
+
+
+                        (
+                            host_id=host_id,
+                            service_id=service_id,
+                            name="priority",
+                            text=text.encode('ascii', 'ignore'))
+
+                    elif result.get(u"type") == u"NS": # Name server record
+                        ns_record = result.get(u"data")[0]
+                        self.createAndAddServiceToInterface(
+                            host_id=host_id,
+                            interface_id=interface_id,
+                            name=ns_record,
+                            protocol="DNS",
+                            ports=[53],
+                            description="DNS Server")
+
+                    elif result.get(u"type") == u"SOA": # Start of Authority Record
+                        ns_record = result.get(u"data")[0] # primary namer server
+                        responsible_party = result.get(u"data")[1] # responsible party of domain
+                        timestamp = result.get(u"data")[2]
+                        refresh_zone_time = result.get(u"data")[3]
+                        retry_refresh_time = result.get(u"data")[4]
+                        upper_limit_time = result.get(u"data")[5]
+                        negative_result_ttl = result.get(u"data")[6]
+
+                        service_id = self.createAndAddServiceToInterface(
+                            host_id=host_id,
+                            interface_id=interface_id,
+                            name=ns_record,
+                            protocol="DNS",
+                            ports=[53],
+                            description="Authority Record")
+
+                        text = (
+                            "Responsible Party: " + responsible_party +
+                            "\nTimestep: " + timestamp +
+                            "\nTime before zone refresh (sec): " + refresh_zone_time +
+                            "\nTime before retry refresh (sec): " + retry_refresh_time +
+                            "\nUpper Limit before Zone is no longer authoritive (sec): " + upper_limit_time +
+                            "\nNegative Result TTL: " + negative_result_ttl)
+
+                        self.createAndAddNoteToService(
+                            host_id=host_id,
+                            service_id=service_id,
+                            name="priority",
+                            text=text.encode('ascii', 'ignore'))
+
+                    elif result.get(u"type") == u"TXT": # TXT record
+                        text = " ".join(result.get(u"data")[:])
+                        self.createAndAddNoteToHost(
+                            host_id=host_id,
+                            name="TXT Information",
+                            text=text.encode('ascii', 'ignore'))
+
+        except:
+            print("some part of the dig plug-in caused an error! Please check repo/dig/plugin.py")
```

Comment:
I think it's a good idea to use logger here.
Check and example here: https://github.com/infobyte/faraday/blob/master/plugins/repo/impact/plugin.py

```python
import logging
logger = logging.getLogger(__name__)
```

Catch the exception and log it with:

```python
except Exception as ex:
      logger.exception(ex)
```

Using the logger will save the error on the faraday logs located at ~/.faraday/logs

Reply:
just added logger to newest commit (2f85cea)